aspnetcore/fundamentals/http-requests.md

```diff
@@ -114,6 +115,38 @@ The configuration for a typed client can also be specified during its registrati
 
 :::code language="csharp" source="http-requests/samples/6.x/HttpRequestsSample/Snippets/Program.cs" id="snippet_AddHttpClientTypedInline":::
 
+### Named and typed clients
+
+Named clients and typed clients have their own strengths and weaknesses. There is a way to combine these two client types to get the best of both worlds.
+
+The primary use case is the following: You want to register the exact same type client multiple times but with different settings (different base address, timeout, and credentials for example).
```

Comment:
In general, I believe this is the case where you should use just an _ordinary Transient service_ instead of a typed client.

That service could either

1) use IHttpClientFactory and incapsulate multiple named clients inside

<details>

```c#
service.UseNamedClient();
service.UseOtherClient();

/* Output:
Name=NamedClient, BaseAddress=https://some-uri/
Name=OtherNamedClient, BaseAddress=https://other-uri/
*/

class Service(IHttpClientFactory _httpClientFactory /*, ... */)
{
    private void UseHttpClient(string name) => Console.WriteLine($"Name={name}, BaseAddress={_httpClientFactory.CreateClient(name).BaseAddress}");
    public void UseNamedClient() => UseHttpClient("NamedClient");
    public void UseOtherClient() => UseHttpClient("OtherNamedClient");
}
```

</details>

-OR-

2) Have an instance of HttpClient passed as a parameter to the method each time

<details>

```c#
foreach (var name in new[] {"NamedClient", "OtherNamedClient" })
{
    service.UseHttpClient(name, _httpClientFactory.CreateClient(name));
}

/* Output:
Name=NamedClient, BaseAddress=https://some-uri/
Name=OtherNamedClient, BaseAddress=https://other-uri/
*/

class Service(/* ... */)
{
    private void UseHttpClient(string name, HttpClient client) => Console.WriteLine($"Name={name}, BaseAddress={client.BaseAddress}");
}
```

</details>

-OR-

3) Leverage Keyed Services DI Pattern (and it can even be non-transient in such case):

```c#
using Microsoft.Extensions.DependencyInjection;

var services = new ServiceCollection();

services.AddHttpClient("NamedClient", c => c.BaseAddress = new Uri("https://some-uri"));
services.AddHttpClient("OtherNamedClient", c => c.BaseAddress = new Uri("https://other-uri"));

services.AddKeyedSingleton<Service>(KeyedService.AnyKey); // instead of a typed client
services.AddSingleton<TwoNamesDependentService>();

var provider = services.BuildServiceProvider();

var dependentService = provider.GetRequiredService<TwoNamesDependentService>();
dependentService.UseServices();

/* Output:
Name=NamedClient, BaseAddress=https://some-uri/
Name=OtherNamedClient, BaseAddress=https://other-uri/
*/

class Service([ServiceKey] string _name, IHttpClientFactory _httpClientFactory)
{
    private HttpClient Client => _httpClientFactory.CreateClient(_name);
    public void UseHttpClient() => Console.WriteLine($"Name={_name}, BaseAddress={Client.BaseAddress}");
}

class TwoNamesDependentService([FromKeyedServices("NamedClient")] Service _service, [FromKeyedServices("OtherNamedClient")] Service _otherService)
{
    public void UseServices()
    {
        _service.UseHttpClient();
        _otherService.UseHttpClient();
    }
}
```